src/Calendar.jsx

```diff
@@ -304,6 +308,12 @@ let Calendar = React.createClass({
   },
 
   @widgetEditable
+  changeCurrentDate(date){
+    var inRangeDate = this.inRangeValue(date ? new Date(date) : this.props.currentDate)
+    notify(this.props.onCurrentDateChange, inRangeDate)
```

Comment:
I would add an equality check here, so you don't trigger an update when the value hasn't changed in a relevant way, probably: `dates.eq(inRangeDate, dateOrNull(this.props.currentDate), VIEW_UNIT[view])`